File: CSharp/005_Program/Program.cs
// File Search
using System;
using System.IO;
class Program {
    static void Main() {
        Console.Write("Directory: ");
        string dir = Console.ReadLine();
        Console.Write("Pattern (*.txt): ");
        string pattern = Console.ReadLine();
        var files = Directory.GetFiles(dir, pattern, SearchOption.AllDirectories);
        Console.WriteLine($"\nFound {files.Length} files");
        foreach (var f in files) Console.WriteLine(f);
    }
}

File: Python/005_Program/program.py
#!/usr/bin/env python3
# File Search - Search files by pattern
import os
import fnmatch

def main():
    print("=== File Search ===")
    directory = input("Directory: ")
    pattern = input("Pattern (e.g., *.txt): ")
    
    matches = []
    for root, dirs, files in os.walk(directory):
        for filename in fnmatch.filter(files, pattern):
            matches.append(os.path.join(root, filename))
    
    print(f"\nFound {len(matches)} files:")
    for match in matches:
        print(match)

if __name__ == "__main__":
    main()
